### ISSUE TITLE
Jxl bit depth

### ISSUE THREAD

**Issue by SimplyCorbett** [libvips/libvips#4083]
Can you expose bit depth on vips for jxl? It currently doesn't expose it but does for avif.

**Comment by jcupitt** [comment]
Hi @SimplyCorbett, I'm planning some work on the jxl connector in September, let's do this at the same time.

**Comment by SimplyCorbett** [comment]
Thanks!

**Comment by SimplyCorbett** [comment]
Also I’m not sure if progressive encoding is enabled by default but if that option can be added as well it would be excellent.

**Comment by jcupitt** [comment]
Yes, that's on the TODO list as well, we have some EU money to help fund development:

https://nlnet.nl/project/libvips/

**Comment by jcupitt** [comment]
There's a PR for jxlsave bitdepth! Finally.

It seems to work, though jxlload will need a matching change to left-shift low-bitdepth images, as per libvips convention.

**Comment by jcupitt** [comment]
> Also I’m not sure if progressive encoding is enabled by default but if that option can be added as well it would be excellent.

I think JXLs are always progressive, the `-p` option in their encoder just makes them even more progressive. So (hopefully?) libvips doesn't need to do anything here.

Since we now have a working PR for bitdepth, I'll close this issue. Let's continue any discussion in the PR.
